applications: nrf_desktop: Add configuration for nrf52833dk_nrf52820 (#6512)
nrfconnect/sdk-nrf: scripts/hid_configurator/README.rst

The script looks for nRF Desktop devices connected to the host through USB, Bluetooth®, or nRF Desktop dongle.
The devices are identified based on Vendor ID.

> If you want to connect to your device which has different Vendor or Product ID you can run script with

Reviewer: Move this under: `Debian/Ubuntu/Linux Mint` and merge it with the current description. This only affects Linux setup actually.

> root permissions on Linux or add new entry to 99-hid.rules file with your Vendor and Product ID.

Reviewer: btw. one sentence per line

       In versions earlier than 5.44, the HID device attached by BlueZ could obtain wrong VID and PID values (ignoring values in Device Information Service), which would stop HIDAPI from opening the device.
       In versions earlier than 5.56, the HID device attached by BlueZ might provide incomplete HID feature report on get operation.

> #. Additionally, to call the Python script on Linux without root rights, copy the provided udev rule in file :file:`99-hid.rules` to :file:`/etc/udev/rules.d` and replugging the device.

Author: I will change it a bit, because you might want to run script without root rights even if you have them (99% situations would probably be like that)

>   If you want to connect to your device which has different Vendor or Product ID you can run script with root permissions or add new entry to :file:`99-hid.rules` file with your Vendor and Product ID and install it by coping it to :file:`/etc/udev/rules.d` and replugging the device.

Reviewer: `... has different Vendor or Product ID than the ones specified in the file ...`

>    If you want to connect to your device which has different Vendor or Product ID than the ones specified in the file you can run script with root permissions or add new entry to :file:`99-hid.rules` file with your Vendor and Product ID and install it by coping it to :file:`/etc/udev/rules.d` and replugging the device.

Reviewer 2: ```suggestion
#. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, you can either run the script if you have the root permission or add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID. 
Then you can copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.
```

> #. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, you can either run the script with the root permission or add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID, then copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d`  and reconnect the device.

Reviewer 2: ```suggestion
#. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, you can either run the script with the root permission or complete the following step without root permission:
    
    a. Add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID.
    #. Copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d`.
    #. Reconnect the device.
    
    Vendor and Product ID can be specified in the configuration file related to the nRF Desktop application.
    The following example shows the entry to add to the :file:`99-hid.rules` file: 
    
    .. parsed-literal::
       :class: highlight

       ATTRS{idVendor}=="my Vendor ID", ATTRS{idProduct}=="my Product ID", MODE="0666", SYMLINK+="nrf52-desktop-my-app-name"
```
After offline discussion with greg, concluded that it a good idea to break down the long sentences into steps.

>    Vendor and Product ID can be defined in config file related to your nrf_desktop application.

Reviewer 2: ```suggestion
Vendor and Product ID can be specified in the configuration file related to the nRF Desktop application.
The following example shows the entry to add to the :file:`99-hid.rules` file: 
   
 .. parsed-literal::
    :class: highlight

     ATTRS{idVendor}=="my Vendor ID", ATTRS{idProduct}=="my Product ID", MODE="0666", SYMLINK+="nrf52-desktop-my-app-name"
```

       In versions earlier than 5.44, the HID device attached by BlueZ could obtain wrong VID and PID values (ignoring values in Device Information Service), which would stop HIDAPI from opening the device.
       In versions earlier than 5.56, the HID device attached by BlueZ might provide incomplete HID feature report on get operation.

> #. In addition, if you don't want to use root access to run the python script, copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.

Reviewer 2: ```suggestion
#. In addition, if you don't want to use the root access to run the python script, copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.
```

> #. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, you can either run the script if you have the root permission or add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID. 

Reviewer 2: ```suggestion
   #. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file , you can either run the script if you have root permission or add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID, then copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d`  and reconnect the device.
```

Reviewer 2: Just thought that if we give it as a separate sentence it will be applicable for both either and or case which is not true. So, had to rephrase it.

>    Then you can copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.

Reviewer 2: ****
```suggestion
```

Reviewer 2: Please remove the blank space as well.

       In versions earlier than 5.44, the HID device attached by BlueZ could obtain wrong VID and PID values (ignoring values in Device Information Service), which would stop HIDAPI from opening the device.
       In versions earlier than 5.56, the HID device attached by BlueZ might provide incomplete HID feature report on get operation.

#. In addition, if you don't want to use the root access to run the python script, copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.

> #. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, you can either run the script with the root permission or complete the following step without root permission:

Reviewer: Can we copy the file to the mentioned directory without root permission?

    a. Add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID.
    #. Copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d`.
    #. Reconnect the device.

    Vendor and Product ID can be specified in the configuration file related to the nRF Desktop application.
    The following example shows the entry to add to the :file:`99-hid.rules` file:

    .. parsed-literal::
       :class: highlight

>        ATTRS{idVendor}=="my Vendor ID", ATTRS{idProduct}=="my Product ID", MODE="0666", SYMLINK+="nrf52-desktop-my-app-name"

Reviewer: Consider adding 2 entries here - for USB and Bluetooth (you can even directly use PID of one of our devices to make it more clear and explicit).

Author: I prefer adding with fields witch are supposed to be changed being called my_xyz so that user knows that it is not needed to change other fields like Mode. "Ooo 666 looks bad I will change it to 111 and then, why it doesn't work :("

Reviewer: Ok, then we can leave it. User could look at existing file to take a look at an example.

Reviewer: `SYMLINK+="nrf52-desktop-my-dev-name"`?

       In versions earlier than 5.44, the HID device attached by BlueZ could obtain wrong VID and PID values (ignoring values in Device Information Service), which would stop HIDAPI from opening the device.
       In versions earlier than 5.56, the HID device attached by BlueZ might provide incomplete HID feature report on get operation.

> #. In addition, if you don't want to use the root access to run the python script, copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.

Reviewer 3: ```suggestion
#. If you don't want to use the root access to run the python script, copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d` and reconnect the device.
```

> #. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, you can either run the script with the root permission or complete the following step to run the script without root permission:

Reviewer 3: ```suggestion
#. If you want to connect to a device with a different Vendor or Product ID other than the one specified in the file, use one of the following options:

   * Run the script with the root permission.
   * Complete the following steps to run the script without root permission:
```

    a. Add a new entry to the :file:`99-hid.rules` file with your Vendor and Product ID.
    #. Copy the provided udev rule from the :file:`99-hid.rules` file to the :file:`/etc/udev/rules.d`.

>     #. Reconnect the device.

Reviewer 3: Indent these lines, so that they start at the height of "C" in "Complete the following steps" above.

    Vendor and Product ID can be specified in the configuration file related to the nRF Desktop application.
    The following examples shows the entry to add to the :file:`99-hid.rules` file to add device connected with USB and Bluetooth:

    .. parsed-literal::
       :class: highlight
       Device connected using USB:
       ATTRS{idVendor}=="my Vendor ID", ATTRS{idProduct}=="my Product ID", MODE="0666", SYMLINK+="nrf52-desktop-my-app-name"
       Device connected using Bluetooth:

>        ATTRS{name}=="my device name", SUBSYSTEMS=="input", MODE="0666", SYMLINK+="nrf52-desktop-my-app-name"

Reviewer: my Bluetooth device name ? To clearly inform that this one must be consistent

Reviewer: `SYMLINK+="nrf52-desktop-my-dev-name"`?
